spec/docx/smoke/api_paragraph_spec.rb

```diff
@@ -314,7 +314,12 @@ def check_caption(element, text_arr)
 
       caption = docx.elements[8]
       expect(caption.nonempty_runs.size).to eq(4)
-      check_caption(caption, ['1', ':', '1', ' caption text '])
+      # TODO: 'check after release'
+      if builder.version >= 'v8.2'
```

Comment:
I'm not quite sure this comparision works as expected

Try this - change this line to something like 

```
if builder.version >= 'v8.9'
```
and execute this code - the if will be `true` still, because is string comparison

Please search in history of codebase of this project - I think mechanism of version comparison was little bit complicated, but I don't quite remember how it looked like

Or maybe I'm mistaken, but still worth the check :)

Reply:
I'm not sure I understand what's wrong.
We only need this check for the current version, and after that we planned to remove it
And if we compare versions like 8.1 or 8.2, it works correctly
![Снимок экрана от 2024-08-03 16-00-26](https://github.com/user-attachments/assets/8e1cfb4f-ab97-4593-a81c-f996987042b5)

Reply:
This is becuase v8.2 and v8.3 are not edge scenarios

When you call >= in your example - you compare the string by symbols and in your example your symbols indeed have bigger and smaller value

But take this example:
```
0> builder.version
=> v8.1.0.169 sdk version:  8.1.0 (build:169) 

0> builder.version >= 'v10.0'
=> true
```
This clearly incorrect result, because v10.0 is newer v8.1.0.169

This is because first two symbols of `v8` is greater that `v1`

Please create a issue for that - this case will create a lot of confusion in test for sure as soon as product reach version v10